R/postprocess.R

```diff
@@ -403,3 +382,77 @@ enw_quantiles_to_long <- function(posterior) {
   long[, quantile := as.numeric(quantile) / 100]
   return(long[])
 }
+
+#' Build the ord_obs `data.table`.
+#'
+#' @param obs Observations as pulled from nowcast$latest[[1]].
+#' @param max_delay Whole number representing the maximum delay
+#' in units of the timestep.
+#' @param internal_timestep The internal timestep in days.
+#' @param timestep The timestep to be used. This can be a string
+#' ("day", "week", "month") or a numeric whole number representing
+#' the number of days.
+#' @param sample String, "get_sample" or "no_sample".
+#' @param nowcast If getting posterior samples, the fit to pull
+#' the draws from.
+#'
+#' @return ord_obs A `data.table`.
+#'
+#' @family postprocess
+
+build_ord_obs <- function(obs, max_delay, internal_timestep, timestep, sample, nowcast = NULL) { # nolint
+  ord_obs <- coerce_dt(
+    obs, required_cols = c("reference_date", "confirm"), group = TRUE
+  )
+  check_timestep_by_group(
+    ord_obs, "reference_date", timestep, exact = TRUE
+  )
+
+  ord_obs <- subset_obs(ord_obs, max_delay, internal_timestep,
+                        select = "after")
+
+sample <- rlang::arg_match(sample, c("get_sample", "no_sample"))
+
+  data.table::setorderv(ord_obs, c(".group", "reference_date"))
+  if (sample == "get_sample") {
```

Comment:
this arg also needs a arg_match. Potentially we could just do this using the presence or absence of nowcast but perhaps that is a bit dangerous?

Reply:
My preference is to do it that way, checking for presence of `nowcast` but wasn't sure if it would be a bad idea. Do you have strong opinions one way or the other?

Reply:
No I don't. If this was a user facing function and I thought they would be very confused by the auto-magical difference you get from passing a data frame I might push for the second name. We could meet in the middle somewhere and rename this to be a bit more generic but a bit clearer about what happens internally (precisely what a short name that communicates that is is currently a bit beyond me).

Reply:
Modified the function to this but didn't yet come up with a good name. hard to convey concisely, will think on it